app/controllers/comments_controller.rb

```diff
@@ -0,0 +1,25 @@
+class CommentsController < ApplicationController
+  
+  http_basic_authenticate_with name: "user", password: "secret", only: :destroy 
+  
+  def create
+    @article = Article.find(params[:article_id])
```

Comment:
Where does the article_id param come from?

Reply:
`ArticlesController` in independent from `CommentController`.

Clue: What is the URL for this action?

Reply:
Of course! came from the routes file, and `resources`, so `article_id` is taken from the actual URL 'article/:id' and build by params.